### ISSUE TITLE
Cmake integration

### ISSUE THREAD

**Issue by stephomi** [elalish/manifold#476]
Just integrated Manifold and it works great, nice!

---

How are we supposed to integrate Manifold in a C++ cmake project?
I target Android/iOS/Wasm/Desktop.

Usually I always avoid `add_subdirectory(manifold)` as it tends to creep lot of stuffs in the global scope, so I prefer to list all the files to compile and flags directly. Unfortunately the thrust dependency is a bit annoying and I couldn't do it (lot of cuda/cub include nested stuffs, I don't know if the thrust cmake is somewhat running some scripts or something but I couldn't make it work with only includes path). And I don't even use CUDA😢.

Also I couldn't make it work with OMP either (fine with TBB though but that's another dependency, and now I need to check and build tbb for iOS and integrate it with iOS). MacOS/iOS is a bit finicky with the OpenMP flags (I don't what's the issue, I guess thrust or a missing `-Xpreprocessor` maybe). There are probably not many people using OpenMP on iOS so maybe not that big of a deal.

...
So in the end I use `add_subdirectory(manifold)` but it conflicts with my glm library (I also use clipper2, but no issue so far as I'm using the same latest version). Right now I simply update the glm dependency in Manifold and it's working fine, but ideally Manifold could use tighter scope.

One way to do it is to add the following in the manifold/polygon/sdf,etc CMakeLists.txt (too many CMakeLists imo :p), etc:
```cmake
target_include_directories(${PROJECT_NAME} SYSTEM PRIVATE ../third_party/glm)
```
And remove the following (maybe I'm missing something, but what's there to link? It's a header-only lib)
```cmake
target_link_libraries(${PROJECT_NAME} PUBLIC glm)
```
It works fine. Still not perfect as the C++ public API still include glm dependencies and there is no `#ifdef` to guard it.

(not familiar with the discussions tab, so maybe there is already a discussion on similar topic)

**Comment by elalish** [comment]
Thanks for the pointers - honestly I'm no cmake expert, so I rely on users like yourself to show how it can be improved for downstream consumption. Those sound like perfectly reasonable changes - mind opening a PR?

**Comment by stephomi** [comment]
I might do it later but I'm not testing the wasm binding, python thing, etc 
For now I forked it and it works fine for me https://github.com/stephomi/manifold/commit/313ec2824f3fade91d8e0f16fb7a2c7d80533ec9

---
I got an error when manifold tries to build the wasm binding:
```shell
ninja: error: 'lib/manifold/bindings/wasm/manifold.*', needed by 'lib/manifold/bindings/wasm/CMakeFiles/js_deps', missing and no known rule to make it
```
Same error if I use Make instead of Ninja.
I didn't have the issue a few days ago so maybe I updated cmake, emcc or something.. I don't know.
It's not an issue for me as I don't need those bindings.

But this is what I do in my cmake:
```cmake
set(MANIFOLD_TEST OFF CACHE BOOL "" FORCE)
set(MANIFOLD_PYBIND OFF CACHE BOOL "" FORCE)
set(PYBIND11_FINDPYTHON OFF CACHE BOOL "" FORCE)
if(IS_WEB)
    set(MANIFOLD_PAR NONE CACHE BOOL "" FORCE)
elseif(IS_ANDROID)
    set(MANIFOLD_PAR OMP CACHE BOOL "" FORCE) # TODO TBB?
else()
    set(MANIFOLD_PAR TBB CACHE BOOL "" FORCE)
endif()
if(IS_WEB)
    add_compile_definitions(_LIBCUDACXX_HAS_THREAD_API_EXTERNAL)
    set(EMSCRIPTEN 0)
    add_subdirectory(lib/manifold/)
    set(EMSCRIPTEN 1)
else()
    add_subdirectory(lib/manifold/)
endif()
```

Not really a bug report, but OMP is much slower than TBB which is a bit curious, maybe a thing on how thrust handles OMP.

**Comment by elalish** [comment]
Yeah, we've noticed OMP issues - TBB is probably a better default.

Want to throw this into a PR? Then we can see how the CI likes it.

**Comment by stephomi** [comment]
Follow up https://github.com/elalish/manifold/pull/482#issuecomment-1626361063
TLDR: likely my error (should stop using SYSTEM :sweat_smile:)
